fhir-server/src/main/java/au/csiro/pathling/fhirpath/FhirPath.java

```diff
@@ -47,4 +48,25 @@
    */
   boolean isSingular();
 
+  /**
+   * Creates a path that can be used to represent a collection which includes elements from both a
+   * source and a target path.
+   *
+   * @param target the path the merge the invoking path with
+   * @param dataset the {@link Dataset} that can be used to evaluate this path against data
+   * @param expression the FHIRPath expression that represents the result
+   * @param idColumn a {@link Column} within the dataset containing the identity of the subject
+   * resource
+   * @param eidColumn a {@link Column} that represents the unique ID for an element within a
+   * collection
+   * @param valueColumn a {@link Column} within the dataset containing the values of the nodes
+   * @param singular an indicator of whether this path represents a single-valued collection
+   * @param thisColumn for paths that traverse from the {@code $this} keyword, this column refers to
+   * the values in the collection
+   * @return the resulting new {@link NonLiteralPath}
+   */
+  NonLiteralPath mergeWith(FhirPath target, Dataset<Row> dataset, String expression,
+      Column idColumn, Optional<Column> eidColumn, Column valueColumn, boolean singular,
+      Optional<Column> thisColumn);
+
```

Comment:
Add Nonnull annotations?

Reply:
I can't seem to be able to get IntelliJ to tell me about this sort of issue.

Maybe it's because this is an interface...

Reply:
Actually, maybe this would have picked it up (in a kind of indirect way):

<img width="422" alt="image" src="https://user-images.githubusercontent.com/55939/106849632-c2271680-66fe-11eb-86a5-0708e63eebeb.png">

Reply:
Fixed in c6592b2.